### PR TITLE
Json Serialization and Deserialization

### PR DESCRIPTION
**Pull request by BeckySharp**
- [De]Serializes each flavor of OdinsonMatch
 - includes tests

### CODE REVIEW

File: core/src/main/scala/ai/lum/odinson/serialization/JsonSerializer.scala [hunk 1]
```diff
@@ -0,0 +1,272 @@
...
+    ujson.Obj(
+      "scalaType" -> n.getClass.getCanonicalName,
+      "name" -> n.name,
+      "label" -> stringOrNull(n.label),
+      "capturedMatch" -> jsonify(n.capturedMatch)
+    )
+  }
+
+  def jsonify(arguments: Map[String, Array[Mention]]): Value = {
+    ujson.Arr(
```
Comment (BeckySharp) [inline core/src/main/scala/ai/lum/odinson/serialization/JsonSerializer.scala:114]: Obj

File: core/src/test/scala/ai/lum/odinson/foundations/TestJsonSerialization.scala [hunk 2]
```diff
@@ -0,0 +1,232 @@
...
+    val m = getMentionFromRule(mentions, "Named")
+    val json = JsonSerializer.jsonify(Array(m))
+    val reconstituted = JsonSerializer.deserialize(json)
+    reconstituted should have length(1)
+
+    mentionsAreEqual(m, reconstituted.head) should be (true)
+  }
+
+  // With state
+  val stateMentions = engine.extractMentions(extractors).toArray
```
Comment (BeckySharp) [inline core/src/test/scala/ai/lum/odinson/foundations/TestJsonSerialization.scala:220]: add state versions here

**Review by BeckySharp — COMMENTED** [review]
todo

File: core/src/main/scala/ai/lum/odinson/IdGetter.scala [hunk 3]
```diff
@@ -7,6 +7,11 @@ trait IdGetter {
   def getSentId: String
 }
 
+class KnownIdGetter(docId: String, sentId: String) extends IdGetter {
```
Comment (marcovzla) [inline core/src/main/scala/ai/lum/odinson/IdGetter.scala:10]: Can you please add a brief comment here?

**Review by marcovzla — CHANGES_REQUESTED** [review]
This looks great, thanks!
I just requested a comment that briefly describes `KnownIdGetter`, but other than that I think this is ready to be merged

File: core/src/main/scala/ai/lum/odinson/IdGetter.scala (hunk 3 above)
Comment (BeckySharp) [reply core/src/main/scala/ai/lum/odinson/IdGetter.scala:14]: addressed, thanks!

**Review by marcovzla — APPROVED** [review]

**Pull request merged**